apps/web/pages/settings/organizations/platform/oauth-clients/components/OAuthClientForm.tsx

```diff
@@ -1,32 +1,61 @@
 import { useCreateOAuthClient } from "@pages/settings/organizations/platform/oauth-clients/hooks/usePersistOAuthClient";
 import { useRouter } from "next/router";
 import type { FC } from "react";
-import React from "react";
-import { useForm } from "react-hook-form";
+import React, { useState, useCallback } from "react";
+import { useForm, useFieldArray } from "react-hook-form";
 
+import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { PERMISSIONS_GROUPED_MAP } from "@calcom/platform-constants/permissions";
 import { showToast } from "@calcom/ui";
-import { Meta, Button, TextField } from "@calcom/ui";
+import { Meta, Button, TextField, Label } from "@calcom/ui";
+import { Plus, Trash } from "@calcom/ui/components/icon";
 
 type FormValues = {
   name: string;
   logo?: string;
-  redirectUri: string;
-  redirectUris: string[];
   permissions: number;
   eventTypeRead: boolean;
   eventTypeWrite: boolean;
   bookingRead: boolean;
   bookingWrite: boolean;
   scheduleRead: boolean;
   scheduleWrite: boolean;
+  appsRead: boolean;
+  appsWrite: boolean;
+  profileRead: boolean;
+  profileWrite: boolean;
+  redirectUris: {
```

Comment:
`useFieldArray` hook only accepts an array of objects that's why had to put in uris inside an object in the array